catalog/nsili/catalog-nsili-source/src/main/java/org/codice/alliance/nsili/source/NsiliSource.java

```diff
@@ -407,11 +411,51 @@ private void getIorStringFromSource() {
             LOGGER.warn("{} : Error retrieving IOR file for {}. {}", id, iorUrl, e.getMessage());
             LOGGER.debug("{} : Error retrieving IOR file for {}.", id, iorUrl, e);
         }
+    }
 
-        if (StringUtils.isNotBlank(iorString)) {
-            LOGGER.debug("{} : Successfully obtained IOR file from {}", getId(), iorUrl);
-        } else {
-            LOGGER.error("{} : Received an empty or null IOR String.", id);
+    /**
+     * Uses FTPClient to obtain the IOR string from the provided URL via FTP.
+     */
+    private void getIorStringFromFtpSource() {
+        URI uri = null;
+        try {
+            uri = new URI(iorUrl);
+        } catch (URISyntaxException e) {
+            LOGGER.error("Invalid URL specified for IOR string: {}", iorUrl);
+        }
+
+        FTPClient ftpClient = new FTPClient();
+        try {
+            if (uri.getPort() > 0) {
+                ftpClient.connect(uri.getHost(), uri.getPort());
+            } else {
+                ftpClient.connect(uri.getHost());
+            }
+
+            if (!ftpClient.login(serverUsername, serverPassword)) {
+                LOGGER.error("{} : FTP server log in unsuccessful.", id);
+                LOGGER.debug("{} : FTP server log in unsuccessful.", id);
+            } else {
+                int timeoutMsec = clientTimeout * 1000;
+                ftpClient.setConnectTimeout(timeoutMsec);
+                ftpClient.setControlKeepAliveReplyTimeout(timeoutMsec);
+                ftpClient.setDataTimeout(timeoutMsec);
+
+                ftpClient.enterLocalPassiveMode();
+                ftpClient.setFileType(FTP.BINARY_FILE_TYPE);
+
+                InputStream inputStream = ftpClient.retrieveFileStream(uri.getPath());
+
+                iorString = IOUtils.toString(inputStream, StandardCharsets.ISO_8859_1.name());
+                //Remove leading/trailing whitespace as the CORBA init can't handle that.
+                iorString = iorString.trim();
+            }
+        } catch (IOException e) {
+            LOGGER.error("{} : Unable to process IOR String. {}", id, e.getMessage());
+            LOGGER.debug("{} : Unable to process IOR String.", id, e);
+        } catch (Exception e) {
+            LOGGER.warn("{} : Error retrieving IOR file for {}. {}", id, iorUrl, e.getMessage());
+            LOGGER.debug("{} : Error retrieving IOR file for {}.", id, iorUrl, e);
         }
```

Comment:
This log message is more detailed than the one for IOException. Recommend removing the catch(IOException) as this will catch those as well.